src/ucp/dt/dt_generic.h

```diff
@@ -20,12 +20,21 @@ typedef struct ucp_dt_generic {
 } ucp_dt_generic_t;
 
 
-static inline ucp_dt_generic_t* ucp_dt_generic(ucp_datatype_t datatype)
+#define UCP_DT_IS_GENERIC(_datatype) \
+    (((_datatype) & UCP_DATATYPE_CLASS_MASK) == UCP_DATATYPE_GENERIC)
```

Comment:
This is existing macro which i'd like to remove eventually (move logic to iterator). If we change it to use ucp_datatype_class() it would require either including dt.inl from dt_generic.h (better not include INL from H) or moving ucp_datatype_class to dt.h (also not very nice), IMO better keep it as is for now

Reply:
10x!